src/util/prolog/prolog.test.ts

```diff
@@ -0,0 +1,285 @@
+				const session = new PrologSession("likes(lisa, bob).", 1000);
+				const result = (await session.executeQuery("likes(lisa, bob).")).getResults();
+
+				expect(result).toStrictEqual([["true"], ["false"]]);
+		});
+
+		test("Truth Value (False)", async () => {
+				const session = new PrologSession("likes(lisa, bob).", 1000);
+				const result = (await session.executeQuery("likes(lisa, dieter).")).getResults();
+
+				expect(result).toStrictEqual([["false"]]);
+		});
+
+		test("One Variable One Result", async () => {
+
+				const session = new PrologSession("likes(lisa, bob).", 1000);
+				const result = (await session.executeQuery("likes(lisa, X).")).getResults();
+				expect(result).toStrictEqual([["bob"], ["false"]]);
+
+		});
+
+		test("Two Variables One Result", async () => {
+				const session = new PrologSession("likes(lisa, bob, karl).", 1000);
+				const result = (await session.executeQuery("likes(lisa, X, Y).")).getResults();
+
+				expect(result).toStrictEqual([["bob", "karl"], ["false"]]);
+
+		});
+
+		test("Three Variables One Result", async () => {
+				const session = new PrologSession("likes(lisa, bob, karl, friedrich).", 1000);
+				const result = (await session.executeQuery("likes(lisa, X, Y, Z).")).getResults();
+
+				expect(result).toStrictEqual([["bob", "karl", "friedrich"], ["false"]]);
+
+		});
+
+		test("Four Variables One Result", async () => {
+				const session = new PrologSession("likes(lisa, bob, karl, friedrich, dieter).", 1000);
+				const result = (await session.executeQuery("likes(lisa, X, Y, Z, A).")).getResults();
+
+				expect(result).toStrictEqual([["bob", "karl", "friedrich", "dieter"], ["false"]]);
+
+		});
+
+		test("One Variable Two Results", async () => {
+
+				const session = new PrologSession("likes(lisa, bob). likes(lisa, dieter).", 1000);
+				const result = (await session.executeQuery("likes(lisa, X).")).getResults();
+				expect(result).toStrictEqual([["bob"], ["dieter"], ["false"]]);
+
+		});
+
+		test("Two Variables Two Results", async () => {
+				const session = new PrologSession("likes(lisa, bob, karl). likes(lisa, karl, bob).", 1000);
+				const result = (await session.executeQuery("likes(lisa, X, Y).")).getResults();
+
+				expect(result).toStrictEqual([["bob", "karl"], ["karl", "bob"], ["false"]]);
+
+		});
+
+		test("Three Variables Two Results", async () => {
+				const session = new PrologSession("likes(lisa, bob, karl, friedrich). likes(lisa, bob, friedrich, karl).", 1000);
+				const result = (await session.executeQuery("likes(lisa, X, Y, Z).")).getResults();
+
+				expect(result).toStrictEqual([["bob", "karl", "friedrich"], ["bob", "friedrich", "karl"], ["false"]]);
+
+		});
+
+		test("Four Variables Two Results", async () => {
+				const session = new PrologSession(`
+					likes(lisa, bob, karl, friedrich, dieter).
+					likes(lisa, bob, karl, dieter, friedrich).
+					`, 1000);
+				const result = (await session.executeQuery("likes(lisa, X, Y, Z, A).")).getResults();
```

Comment:
Ir reads a little bit strange that you call it `result`, even though the method is called `getResults` and it's actually an array.